### DISCUSSION TITLE
[DX] plugin support at the library level

### DISCUSSION THREAD

**Discussion by AnomalRoil** [FiloSottile/age#480] [UX reports]
My use case is really that of a developer using `age` as a library, so with the recent `age` v1.1.0 release that brought official plugin support, I figured I'd start trying to use plugins. I'm starting slowly, just trying to use the Yubikey-agent plugin for now.

However, as a developer, I'm facing quite a few DX issues, which I'd love to discuss here.
 
My first issue is that it seems that currently the native, exported `ParseIdentities` method is **only supporting X25519 identities**:
https://github.com/FiloSottile/age/blob/8e3f74c283b2e9b3cd0ec661fa4008504e536d20/parse.go#L21-L23

Is there any specific reasons for expecting users of the library not to want plugin support?
Or maybe any guidance as to how this should happen for a library user?

Now, there are unexported methods in your main CLI package that support plugins:
https://github.com/FiloSottile/age/blob/8e3f74c283b2e9b3cd0ec661fa4008504e536d20/cmd/age/parse.go#L198-L210

So I figured, sure enough, I could just copy the required code in my own codebase, but here comes my second issue:
the package `plugin` is currently in the `age` internals: "filippo.io/age/internal/plugin", which means I cannot import it to use `plugin.NewIdentity` in my code.

I could have tried to copy the whole plugin package in turn, but then I would have to also copy the `bech32` and `format` ones...

I realize that plugins are basically just executed as binaries by `age` and so I guess I could as well just invoke the `age` binary directly in my code, but then it's not really acting as a library anymore.

So, what's your take on the plugin part and `age` as a library? It would be nice to easily have hardware token support for many use-cases.

**Comment by ngortheone** [comment]
I have exactly the same problem. I'd love to get plugin support at the library level
